where-to-travel/src/main/webapp/script.js

```diff
@@ -1,212 +1,401 @@
-//This is map stylings for the GMap api
+// This is map stylings for the GMap api
 const mapStyles = [
   {
-    "featureType": "landscape",
-    "stylers": [
+    featureType: 'landscape',
+    stylers: [
       {
-        "visibility": "off"
+        visibility: 'off'
       }
     ]
   },
   {
-    "featureType": "poi",
-    "elementType": "labels.icon",
-    "stylers": [
+    featureType: 'poi',
+    elementType: 'labels.icon',
+    stylers: [
       {
-        "visibility": "off"
+        visibility: 'off'
       }
     ]
   },
   {
-    "featureType": "transit",
-    "stylers": [
+    featureType: 'transit',
+    stylers: [
       {
-        "visibility": "off"
+        visibility: 'off'
       }
     ]
   }
 ];
-//End map stylings
+// End map stylings
 
-const submitId = "submit-id";
-const hoursId = "hrs";
-const minutesId = "mnts";
+// Thresholds for termination of search algorithm
+const placesThreshold = 30;
+const attemptsThreshold = 5;
+const directionThreshold = 5;
+
+// Document ids for user input elements
+const submitId = 'submit';
+const hoursId = 'hrs';
+const minutesId = 'mnts';
 
 let map;
 let home = null;
+let homeMarker = null;
+let markers = [];
 
 // Add gmap js library to head of page
-let script = document.createElement('script');
-script.src = 'https://maps.googleapis.com/maps/api/js?key=' 
-              + secrets['googleMapsKey'] + '&libraries=places';
+const script = document.createElement('script');
+script.src =
+  'https://maps.googleapis.com/maps/api/js?key=' +
+  secrets['googleMapsKey'] +
+  '&libraries=places';
 script.defer = true;
 script.async = true;
 
 document.head.appendChild(script);
 
-/**
- * Initializes map window, runs on load.
- */
+/** Initializes map window, runs on load. */
 async function initialize() {
-  let submit = document.getElementById(submitId);
+  const submit = document.getElementById(submitId);
   submit.addEventListener('click', submitDataListener);
-  home = await getUserLocation();    
-
+ 
   const mapOptions = {
-    center: home, 
+    center: {lat: 36.150813, lng: -40.352239},
```

Comment:
Adding a constant center back to master? shouldn't this be home?

Reply:
This is for when the user first loads the site and hasn't entered a home location yet. This allows for the inital map to be pretty zoomed with the center being the atlantic ocean and being able to seem of north/south america, africa, europe, and some of asia.

Reply:
I understand. Fantastic!

Reply:
This is great! Could you add a comment to explain this?